platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/command/ProfileInstallCommand.java

```diff
@@ -0,0 +1,195 @@
+            printSectionHeading("Installing Features");
+            installFeatures.stream().distinct().forEach(feature -> {
+                printItemStatusPending("Installing: ", feature);
+                executeAsSystem(() -> {
+                    try {
+                        featuresService.installFeature(feature,
+                                NO_AUTO_REFRESH);
+                    } catch (ResolutionException e) {
+                        printItemStatusFailure("Install Failed: ", feature);
+                        throw new ResolutionException(e);
+                    }
+                    return true;
+                });
+                printItemStatusSuccess("Installed: ", feature);
+            });
+        }
+    }
+
+    private void uninstallFeatures(FeaturesService featuresService, List<String> uninstallFeatures) {
+        if (uninstallFeatures != null) {
+            printSectionHeading("Uninstalling Features");
+            uninstallFeatures.stream().distinct().forEach(feature -> {
+                printItemStatusPending("Uninstalling: ", feature);
+                executeAsSystem(() -> {
+                    Feature featureObject = featuresService.getFeature(feature);
+                    try {
+                        featuresService.uninstallFeature(featureObject.getName(),
+                                featureObject.getVersion(),
+                                NO_AUTO_REFRESH);
+                    } catch (ResolutionException e) {
+                        printItemStatusFailure("Uninstall Failed: ", feature);
+                        throw new ResolutionException(e);
+                    }
+                    return true;
+                });
+                printItemStatusSuccess("Uninstalled: ", feature);
+            });
+        }
+    }
+
+    private void stopBundles(BundleService bundleService, List<String> stopBundleNames) {
+        if (stopBundleNames != null) {
+            printSectionHeading("Stopping Bundles");
+            stopBundleNames.stream().distinct().forEach(bundle -> {
+                printItemStatusPending("Stopping: ", bundle);
+                executeAsSystem(() -> {
+                    try {
+                        bundleService.getBundle(bundle).stop();
+                    } catch (BundleException e) {
+                        printItemStatusFailure("Stop Failed: ", bundle);
+                        throw new BundleException("Stop Failed for bundle", e);
+                    }
+                    return true;
+                });
+                printItemStatusSuccess("Stopped: ", bundle);
+            });
+        }
+    }
+
+    private <T> T executeAsSystem(Callable<T> func) {
+        Subject systemSubject = getSystemSubject();
+        LOGGER.debug("System Subject retrieved: " + SubjectUtils.getName(systemSubject));
+        if (systemSubject == null) {
+            throw new RuntimeException("Could not get system user to auto install applications. ");
```

Comment:
Similar to below, I think I borrowed this from another command that used the subject